### PR TITLE
PR into master from dev/olga/4597-MMD-it-Highlight-introduces-duplication-and-has-inconsistent-highlighting

### PR DESCRIPTION
**Pull request by OlgaRedozubova**
branch: dev/olga/4597-MMD-it-Highlight-introduces-duplication-and-has-inconsistent-highlighting

### Why:
Fixed MMD-it Highlight introduces duplication and has inconsistent highlighting [#4597](https://github.com/Mathpix/monorepo/issues/4597)

### What's being changed:

1. Added `mergingHighlights` function to merge highlights.
Screen Shot 2023-06-12 at 21 33 55

2. Added tests for it

3. Fixed inline_math highlighting:

Screen Shot 2023-06-12 at 21 36 38

Screen Shot 2023-06-12 at 21 40 30

### CODE REVIEW

File: src/markdown/highlight/common.ts [hunk 1]
```diff
@@ -4,6 +4,34 @@ export const sortHighlights = (a, b) => {
   return a.start > b.start ? 1 : -1
 };
 
+export const mergingHighlights = (highlights) => {
```
Comment (iammosespaulr) [inline src/markdown/highlight/common.ts:7]: I'm not totally sure that MMD should be handling 'merging' highlights though.

**Review by iammosespaulr — APPROVED** [review]
Awesome!

File: src/markdown/highlight/common.ts (hunk 1 above)
Comment (OlgaRedozubova) [reply src/markdown/highlight/common.ts:7]: I've just changed mergingHighlights function
Now, adjacent highlights will not be merged.

Comment (iammosespaulr) [reply src/markdown/highlight/common.ts:7]: Thanks @OlgaRedozubova!

**Pull request merged**